src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -375,8 +370,8 @@ internal void CheckCollectionValid()
 
         private unsafe void ReleaseCollection()
         {
-            Interlocked.Exchange(ref _c4coll, IntPtr.Zero);
-            Native.c4coll_release((C4Collection*)_c4coll);
+            var old = Interlocked.Exchange(ref _c4coll, IntPtr.Zero);
+            Native.c4coll_release((C4Collection*)old);
             _c4coll = IntPtr.Zero;
```

Comment:
This is redundant, it is handled by the first line.  That is what Interlocked.Exchange does, both set a value and return its old one in an atomic way.